public/js/admin/donateDom.js

```diff
@@ -0,0 +1,104 @@
+const infoBtn = document.querySelectorAll('.tap--personal');
+const donateBtn = document.querySelectorAll('.tap--donation');
+const donateDiv = document.querySelectorAll('.cards--info-donation');
+const infoDiv = document.querySelectorAll('.cards--info-personal');
+const newDonatesSection = document.querySelector('.cards--new');
+const currentDonatesSection = document.querySelector('.cards--current');
+const doneDonatesSection = document.querySelector('.cards--done');
+const newBtn = document.querySelector('.new');
+const currentBtn = document.querySelector('.current');
+const doneBtn = document.querySelector('.done');
+
+const doneBtnCheck = document.querySelector('.doneBtn'); // the btn for the done to convert from current to done
+
+doneBtnCheck.addEventListener('click', (e) => {
+  e.preventDefault();
+  const checkBoxes = document.querySelectorAll('.checkBox'); // select all checkbox input as nodelist
+  // const doneBtnCheckValue = checkBoxes.value; // take the value of the checkboxes
+  const checkBoxesArray = Array.from(checkBoxes);
+  const values = [];
+  checkBoxesArray.forEach((element) => {
+    if (element.checked) {
+      values.push(element.id);
+    }
+  });
+  values.forEach((element) => {
+    const cardInfo = {
+      id: element,
+      obj: {
+        done: 'done',
+      },
+    };
+    fetch('/admin/donates', {
+      method: 'POST',
+      body: JSON.stringify(cardInfo),
+      credentials: 'same-origin',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    }).then(res => console.log(res))
+      .catch(er => console.log(er));
+  });
+});
+
+
+newBtn.addEventListener('click', (e) => {
+  e.preventDefault();
+  newDonatesSection.classList.remove('hide');
+  currentDonatesSection.classList.add('hide');
+  doneDonatesSection.classList.add('hide');
+});
+
+currentBtn.addEventListener('click', (e) => {
+  e.preventDefault();
+  currentDonatesSection.classList.remove('hide');
+  newDonatesSection.classList.add('hide');
+  doneDonatesSection.classList.add('hide');
+});
+
+doneBtn.addEventListener('click', (e) => {
+  e.preventDefault();
+  doneDonatesSection.classList.remove('hide');
+  newDonatesSection.classList.add('hide');
+  currentDonatesSection.classList.add('hide');
+});
+
+
+const infoBtnArray = Array.from(infoBtn);
+const donateBtnArray = Array.from(donateBtn);
+const infoDivArray = Array.from(infoDiv);
+const donateDivArray = Array.from(donateDiv);
+
+infoBtnArray.forEach((element, index) => {
+  element.addEventListener('click', (e) => {
+    e.preventDefault();
+    infoDivArray[index].classList.remove('hide');
+    donateDivArray[index].classList.add('hide');
+  });
+});
+
+donateBtnArray.forEach((element, index) => {
+  element.addEventListener('click', (e) => {
+    e.preventDefault();
+    infoDivArray[index].classList.add('hide');
+    donateDivArray[index].classList.remove('hide');
+    const cardInfo = {
```

Comment:
@AbdallahAmmar96 Line 85-102 here is pretty similar to line 26-41 above.  You could try pulling out the common code into a function in this file.  You could pass in `obj` as the argument to the function, and the function could return the fetch result so that each place that called the function could handle the then / catch that it returns.
That would be a good improvement, to consolidate that duplicated code (then you could fix bugs just in one place instead of both places if you find bugs).